api/analyzers/source_analyzer.py

```
@@ -125,15 +134,16 @@ def second_pass(self, graph: Graph, path: Path) -> None:
             lsps[".py"] = SyncLanguageServer.create(config, logger, str(path))
         else:
             lsps[".py"] = NullLanguageServer()
         with lsps[".java"].start_server(), lsps[".py"].start_server():
-            for file_path, file in self.files.items():
-                logging.info(f'Processing file: {file_path}')
+            files_len = len(self.files)
+            for i, (file_path, file) in enumerate(self.files.items()):
+                logging.info(f'Processing file ({i + 1}/{files_len}): {file_path}')
                 for _, entity in file.entities.items():
-                    entity.resolved_symbol(lambda key, symbol: analyzers[file_path.suffix].resolve_symbol(self.files, lsps[file_path.suffix], file_path, key, symbol))
+                    entity.resolved_symbol(lambda key, symbol: analyzers[file_path.suffix].resolve_symbol(self.files, lsps[file_path.suffix], file_path, path, key, symbol))
```

Ruff (0.8.2)

Function definition does not bind loop variable `file_path`

(B023)